src/Microsoft.PowerShell.Security/security/CertificateProvider.cs

```diff
@@ -3598,81 +3443,25 @@ public List<DnsNameRepresentation> DnsNameList
         /// </summary>
         public DnsNameProperty(X509Certificate2 cert)
         {
-            if (DownLevelHelper.NativeFilteringSupported())
-            {
-                if (cert != null)
-                {
-                    //need to get subject alternative name from the certificate context
-                    _dnsList = GetCertNames(
-                            cert.Handle,
-                            Security.NativeMethods.AltNameType.CERT_ALT_NAME_DNS_NAME);
-                }
-            }
-        }
-
-        // Wrapper function for CCGetCertNameList and CCFreeStringArray
-        private List<DnsNameRepresentation> GetCertNames(IntPtr certHandle, Security.NativeMethods.AltNameType nameType)
-        {
-            DWORD cPunycodeName;
-            IntPtr papwszPunycodeNames = IntPtr.Zero;
-            IntPtr papwszUnicodeNames = IntPtr.Zero;
-            List<DnsNameRepresentation> names = new List<DnsNameRepresentation>();
-            int hr;
-
-            if (certHandle != IntPtr.Zero)
+            _dnsList = new List<DnsNameRepresentation>();
+            foreach (X509Extension extension in cert.Extensions)
             {
-                hr = Security.NativeMethods.CCGetCertNameList(
-                            certHandle,
-                            nameType,
-                            0,              // no conversion to Unicode
-                            out cPunycodeName,
-                            out papwszPunycodeNames);
-                if (hr != Security.NativeConstants.S_OK)
-                {
-                    if (hr != Security.NativeMethods.CRYPT_E_NOT_FOUND)
-                    {
-                        throw Marshal.GetExceptionForHR(hr);
-                    }
-                    cPunycodeName = 0;
-                }
-
-                try
+                if (extension.Oid.FriendlyName == "Subject Alternative Name")
                 {
-                    if (0 < cPunycodeName)
+                    string[] names = extension.Format(true).Split(Environment.NewLine);
+                    foreach(string nameLine in names)
                     {
-                        DWORD cUnicodeName;
-
-                        hr = Security.NativeMethods.CCGetCertNameList(
-                                    certHandle,
-                                    nameType,
-                                    Security.NativeMethods.CryptDecodeFlags.CRYPT_DECODE_ENABLE_IA5CONVERSION_FLAG,
-                                    out cUnicodeName,
-                                    out papwszUnicodeNames);
-                        if (hr != Security.NativeConstants.S_OK)
+                        // Get the part after 'DNS Name='
+                        if(nameLine.Length > 9)
```

Comment:
Yes, I was planning on adding it with the support for punyCode.  The only way I can see puny code being supported is a different record format.